src/shims/posix/sync.rs

```diff
@@ -62,7 +62,10 @@ fn mutex_get_kind<'mir, 'tcx: 'mir>(
     mutex_op: OpTy<'tcx, Tag>,
 ) -> InterpResult<'tcx, ScalarMaybeUninit<Tag>> {
     let offset = if ecx.pointer_size().bytes() == 8 { 16 } else { 12 };
-    ecx.read_scalar_at_offset(mutex_op, offset, ecx.machine.layouts.i32)
+    ecx.read_scalar_at_offset_atomic(
+        mutex_op, offset, ecx.machine.layouts.i32,
+        AtomicReadOp::Acquire
```

Comment:
You added extra vector clocks to model mutex synchronization... so why should these accesses here be release/acquire?  Isn't it it enough to do relaxed accesses? In fact, what goes wrong when using non-atomic accesses? (I imagine something could go wrong but it is not immediately obvious to me.)